lacework/resource_lacework_report_rule.go

```diff
@@ -0,0 +1,576 @@
+	d.Set("email_alert_channels", response.Data.EmailAlertChannels)
+	d.Set("resource_groups", response.Data.Filter.ResourceGroups)
+	setNotificationTypes(d, response.Data.ReportNotificationTypes)
+
+	log.Printf("[INFO] Read report rule with guid %s\n", response.Data.Guid)
+	return nil
+}
+
+func resourceLaceworkReportRuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework       = meta.(*api.Client)
+		resourceGroups = d.Get("resource_groups").(*schema.Set).List()
+		severities     = api.NewReportRuleSeverities(castAttributeToStringSlice(d, "severities"))
+		channels       = d.Get("email_alert_channels").(*schema.Set).List()
+	)
+
+	notifications, err := getReportRuleNotifications(d)
+	if err != nil {
+		return err
+	}
+
+	reportRule, err := api.NewReportRule(d.Get("name").(string),
+		api.ReportRuleConfig{
+			Description:        d.Get("description").(string),
+			Severities:         severities,
+			ResourceGroups:     castStringSlice(resourceGroups),
+			EmailAlertChannels: castStringSlice(channels),
+			NotificationTypes:  notifications,
+		},
+	)
+
+	if err != nil {
+		return err
+	}
+
+	reportRule.Guid = d.Id()
+
+	if !d.Get("enabled").(bool) {
+		reportRule.Filter.Enabled = 0
+	}
+
+	log.Printf("[INFO] Updating report rule with data:\n%+v\n", reportRule)
+	response, err := lacework.V2.ReportRules.Update(reportRule)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.Guid)
+	d.Set("name", response.Data.Filter.Name)
+	d.Set("description", response.Data.Filter.Description)
+	d.Set("guid", response.Data.Guid)
+	d.Set("enabled", response.Data.Filter.Enabled == 1)
+	d.Set("created_or_updated_time", response.Data.Filter.CreatedOrUpdatedTime)
+	d.Set("created_or_updated_by", response.Data.Filter.CreatedOrUpdatedBy)
+	d.Set("type", response.Data.Type)
+
+	log.Printf("[INFO] Updated report rule with guid %s\n", response.Data.Guid)
+	return nil
+}
+
+func resourceLaceworkReportRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	log.Printf("[INFO] Deleting report rule with guid %s\n", d.Id())
+	err := lacework.V2.ReportRules.Delete(d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Deleted report rule with guid %s\n", d.Id())
+	return nil
+}
+
+func importLaceworkReportRule(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	var response api.ReportRuleResponse
+	lacework := meta.(*api.Client)
+
+	log.Printf("[INFO] Importing Lacework Report Rule with guid: %s\n", d.Id())
+
+	if err := lacework.V2.ReportRules.Get(d.Id(), &response); err != nil {
+		return nil, fmt.Errorf(
+			"unable to import Lacework resource. Report Rule with guid '%s' was not found",
+			d.Id(),
+		)
+	}
+	log.Printf("[INFO] Report Rule found with guid: %s\n", response.Data.Guid)
+	return []*schema.ResourceData{d}, nil
+}
+
+func getReportRuleNotifications(d *schema.ResourceData) (api.ReportRuleNotifications, error) {
```

Comment:
why do we return `(api.ReportRuleNotifications, error)` if there is not even one error being returned?